packages/core/src/tile/model/point/point.ts

```diff
@@ -0,0 +1,130 @@
+import { PointCloudMaterial } from '../../materials/pointShaderMaterial';
+import { PointResponse, PointShape, TypedArray } from '../../types';
+import { Tile, UpdateOptions } from '../tile';
+
+export interface PointUpdateOptions extends UpdateOptions<PointResponse> {
+  pointOptions?: UniformBuffer;
+  feature?: Feature;
+  groupUpdate?: { categoryIndex: number; group: number };
+  groupState?: Map<string, Map<number, number>>;
+  pointShape?: PointShape;
+}
+
+export class PointTile extends Tile<PointResponse> {
+  protected material: ShaderMaterial;
+  private groups!: Int32Array;
+  private meshData: Record<string, TypedArray>;
+  private vertexCount: number;
+  private activeFeature?: Feature;
+
+  constructor(scene: Scene, response: PointResponse) {
+    super(scene, response);
+
+    this.material = PointCloudMaterial(this.scene);
+    this.mesh.material = this.material;
+    this.meshData = {};
+    this.vertexCount = 0;
+    this.mesh.layerMask = 0b1;
+    this.mesh.renderingGroupId = 3;
+
+    this.update({ response });
+  }
+
+  public update(updateOptions: PointUpdateOptions): void {
+    if (updateOptions.response) {
+      this.meshData = updateOptions.response.attributes;
+      const vertexData = new VertexData();
+
+      vertexData.positions = updateOptions.response.attributes[
+        'position'
+      ] as Float32Array;
+      vertexData.applyToMesh(this.mesh, false);
+
+      this.vertexCount = vertexData.positions.length / 3;
+      this.groups = new Int32Array(this.vertexCount).fill(0);
+    }
+
+    if (updateOptions.pointOptions) {
+      this.material.setUniformBuffer(
+        'pointOptions',
+        updateOptions.pointOptions
+      );
+    }
+
+    if (updateOptions.pointShape) {
+      this.material.setDefine(
+        'POINT_TYPE',
+        updateOptions.pointShape.toString()
+      );
+    }
+
+    if (updateOptions.feature) {
+      this.material.setDefine(
+        'FEATURE_TYPE',
+        updateOptions.feature.type.toString()
+      );
+      this.activeFeature = updateOptions.feature;
+
+      switch (updateOptions.feature.type) {
+        case FeatureType.FLAT_COLOR:
+          //skip
+          break;
+        case FeatureType.CATEGORICAL:
+          {
+            this.groups.fill(32);
+            this.mesh.setVerticesBuffer(
+              new VertexBuffer(
+                this.scene.getEngine(),
+                this.groups,
+                'group',
+                true,
+                false,
+                1
+              )
+            );
+
+            const state = updateOptions.groupState?.get(
+              updateOptions.feature.name
+            );
+            const categories = this.meshData[
+              updateOptions.feature.name
+            ] as Int32Array;
+
+            if (state) {
+              for (let index = 0; index < this.groups.length; ++index) {
+                this.groups[index] = state.get(categories[index]) ?? 32;
```

Comment:
ΝΙΤ: Personally i'd save **32** to a variable, so the variable name can give a hint what 32 is.